src/components/ModalPin.js

```diff
@@ -79,9 +77,18 @@ export class ModalPin extends React.Component {
 
     // Set the PIN on the instance variable and close the modal.
     this.pin = pin;
+    this.onSuccess();
+  }
+
+  onSuccess() {
+    this.props.onSuccess({ pin: this.pin })
+    this.props.onClose();
+  }
+
+  onCancel(e) {
+    e.preventDefault();
 
-    // Necessary callbacks will be executed at the `onHidden` modal event
-    $('#modalPin').modal('hide');
+    this.props.onClose();
```

Comment:
suggestion: Replace this too with `this.onModalHidden()`. The logic will be the same, as I see it.

In short, centralize all modal closing logic in this function, to simplify it.

Reply:
Done, thanks!